appinventor/buildserver/src/com/google/appinventor/buildserver/util/AARLibrary.java

```diff
@@ -246,6 +246,12 @@ public void unpackToDirectory(final File path) throws IOException {
           throw new IOException("Unable to create directory " + path.getAbsolutePath());
         } else if (!entry.isDirectory()) {
           try {
+            // Need to make sure the parent directory is present. Files can appear
+            // in a ZIP (AAR) file without an explicit directory object
+            File parentDir = target.getParentFile();
+            if (!parentDir.exists()) {
+              parentDir.mkdirs();
```

Comment:
Can we imagine a scenario where the temp directory wouldn't have permission for the buildserver to make parent directories, or where this might otherwise return false (disk out of space)? I guess in either case the server is hosed. We might want to consider throwing an IOException here similar to on line 246.